### ISSUE TITLE
Massive install libraries cause the client to shutdown the plugin.

### ISSUE THREAD

**Issue by ABaumher** [ABaumher/galaxy-integration-steam#34]
GOG seems to give us about 20 seconds of blocking code execution before sending us a `shutdown`. Currently the code tries to synchronously determine this data. It may be possible to fix.

**Comment by ABaumher** [comment]
@engiefox i may have a solution here, but i'll need a minute.

**Comment by don-de-marco** [comment]
Just cross-posting this here:
This code just collects some file paths and is done quickly. The most work is done in get_game_size below that where for each game the corresponding state file (these .vcf files in your steamapps folders) is parsed and the size returned. There's no heavy lifting going on and the GIL only is of concern when there's lots of processing, which we do not have here either. IMO it's just the I/O that's hitting us hard because we need to read a couple of hundreds of files. And that just takes time, using executors would make it even worse because of the overhead introduced by using them

**Comment by GabrielAdrianAlves** [comment]
I'm having an issue with the sync since I added a shared family library. My error message is this:

"steam_network.cache_proto - INFO - Timed out waiting for games cache to get ready"

Last player works, list of games works. But achievements and total played time are not updating. My library now with the shared family library has 3281 games. I understand it's not a common library size, but is it possible to make it work?

**Comment by ABaumher** [comment]
We're working on something, but that's a different issue than the one here. You can try our latest test build, which another user with a large library has been testing. I'll see if i can get a zipped version so you don't have to do the install process manually.

**Comment by ABaumher** [comment]
@GabrielAdrianAlves give this a try. This is based off of Version 1.0.5, with a bevy of improvements. We just merged in the patches from 1.0.6 and 1.0.7, and it is possible we messed up something in this process, so if you hit bugs, let us know. 

[steam_v1.1.0_alpha.zip](https://github.com/ABaumher/galaxy-integration-steam/files/11837636/steam_v1.1.0.zip)

**Comment by GabrielAdrianAlves** [comment]
Awesome thank you @ABaumher . I will give it a try and report back any outstanding findings in #17 .

For an update, today the played times of the games I've played recently updated, but the achievements didn't. Maybe it's just taking a really long time because of how big the library is now.

**Comment by don-de-marco** [comment]
yeah, achievements sync just takes ages for larger libraries. unfortunately, currently there's nothing much we can do about it besides rewriting the whole code that does the import
